src/interface/src/app/services/session.service.ts

```diff
@@ -1,19 +1,50 @@
-import { BehaviorSubject } from 'rxjs';
-import { Region } from '../types/region.types';
+import { BehaviorSubject, interval, Observable } from 'rxjs';
 import { Injectable } from '@angular/core';
+import { MapConfig, MapViewOptions, Region } from '../types';
+
+/** How often the user's session should be saved to local storage (in ms). */
+const SESSION_SAVE_INTERVAL = 60000;
 
 /**
  * The session service keeps track of where the guest or logged-in user left
  * off.
  */
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class SessionService {
-  readonly region$ = new BehaviorSubject<Region|null>(null);
+  /** Components can subscribe to this Observable to perform operations repeatedly
+   *  at a set interval (like saving state). */
+  readonly sessionInterval$: Observable<number> = interval(
+    SESSION_SAVE_INTERVAL
+  );
+
+  readonly mapConfigs$ = new BehaviorSubject<MapConfig[] | null>(null);
+  readonly mapViewOptions$ = new BehaviorSubject<MapViewOptions | null>(null);
+  readonly region$ = new BehaviorSubject<Region | null>(null);
 
   constructor() {
-    this.region$.next(localStorage.getItem('region') as Region|null);
+    let storedMapConfigs = localStorage.getItem('mapConfigs');
```

Comment:
Use "const" if storedMapConfigs and storedMapViewOptions are not reassigned

Reply:
done!